core/consensus/babe/impl/babe_synchronizer_impl.cpp

```diff
@@ -3,138 +3,858 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include "consensus/babe/impl/babe_synchronizer_impl.hpp"
-
+#include "babe_synchronizer_impl.hpp"
+#include <blockchain/block_tree_error.hpp>
 #include <random>
 
-#include <boost/assert.hpp>
-#include "blockchain/block_tree_error.hpp"
-#include "common/visitor.hpp"
-#include "network/protocols/sync_protocol.hpp"
-#include "primitives/block.hpp"
+#include "network/types/block_attributes.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::consensus, BabeSynchronizerImpl::Error, e) {
+  using E = kagome::consensus::BabeSynchronizerImpl::Error;
+  switch (e) {
+    case E::SHUTTING_DOWN:
+      return "Node is shutting down";
+    case E::EMPTY_RESPONSE:
+      return "Response is empty";
+    case E::RESPONSE_WITHOUT_BLOCK_HEADER:
+      return "Response does not contain header of some block";
+    case E::RESPONSE_WITHOUT_BLOCK_BODY:
+      return "Response does not contain body of some block";
+    case E::DISCARDED_BLOCK:
+      return "Block is discarded";
```

Comment:
Is it possible to also provide the reason in the error message?

Reply:
More details are provided by logging BlockExecutor. For this case any details do not matter - it means that block and all his descendant are discarded.